docs/pip.md

```diff
@@ -31,6 +31,9 @@ JSON input:
 ```jsonc
 {
   "type": "pip",
+  // allow to use binary packages (wheels)
+  // defaults to true
+  "allow_binary": true,
```

Comment:
I would leave the docs update out for now, there is a separate story for that

And we might want to reconsider what the default should be, given that wheels break security and sorce compliance requirements